source/app/libpd.c

```diff
@@ -173,7 +174,22 @@ static void parodus_receive()
                     {
 						memset(res_wrp_msg, 0, sizeof(wrp_msg_t));
                         getCurrentTime(startPtr);
-                        processRequest((char *)wrp_msg->u.req.payload, wrp_msg->u.req.transaction_uuid, ((char **)(&(res_wrp_msg->u.req.payload))));
+			headers_t *res_headers;
+                        res_headers = (headers_t *)malloc(sizeof(headers_t));
+			if(res_headers != NULL) {
```

Comment:
if null, error logging and return needs to be done

Reply:
added the changes